test/libsolidity/SyntaxTest.cpp

```diff
@@ -60,22 +81,39 @@ SyntaxTest::SyntaxTest(string const& _filename)
 
 bool SyntaxTest::run(ostream& _stream, string const& _linePrefix, bool const _formatted)
 {
+	char const versionPragma[] = "pragma solidity >=0.0;\n";
+	int const versionPragmaLength = strlen(versionPragma);
 	m_compiler.reset();
-	m_compiler.addSource("", "pragma solidity >=0.0;\n" + m_source);
+	m_compiler.addSource("", versionPragma + m_source);
 	m_compiler.setEVMVersion(dev::test::Options::get().evmVersion());
 
 	if (m_compiler.parse())
 		m_compiler.analyze();
 
 	for (auto const& currentError: filterErrors(m_compiler.errors(), true))
-		m_errorList.emplace_back(SyntaxTestError{currentError->typeName(), errorMessage(*currentError)});
+	{
+		int locationStart = -1, locationEnd = -1;
+		if (auto location = boost::get_error_info<errinfo_sourceLocation>(*currentError))
+		{
+			if (location->start >= versionPragmaLength)
```

Comment:
Perhaps a comment here could help: ignore the version pragma inserted by the testing tool when calculating locations.